Partial tests are failing for feedreader

### The problem

When using partial CI, `feedreader` tests are failing.
```
===================================================================== short test summary info ======================================================================
FAILED tests/components/feedreader/test_init.py::test_feed_default_max_length[pyloop] - assert 15 == 20
FAILED tests/components/feedreader/test_init.py::test_feed_without_publication_date_and_title[pyloop] - assert 0 == 3

Results (4.96s):
       9 passed
       2 failed
         - tests/components/feedreader/test_init.py:160 test_feed_default_max_length[pyloop]
         - tests/components/feedreader/test_init.py:182 test_feed_without_publication_date_and_title[pyloop]
```

### What version of Home Assistant Core has the issue?

core-2022.02.0.dev0

### What was the last working version of Home Assistant Core?

_No response_

### What type of installation are you running?

Home Assistant Core

### Integration causing the issue

feedreader

### Link to integration documentation on our website

_No response_

### Example YAML snippet

_No response_

### Anything in the logs that might be useful for us?

_No response_

### Additional information

_No response_

--- a/homeassistant/components/feedreader/__init__.py
+++ b/homeassistant/components/feedreader/__init__.py
@@ -9,8 +9,10 @@ import feedparser
 import voluptuous as vol
 
 from homeassistant.const import CONF_SCAN_INTERVAL, EVENT_HOMEASSISTANT_START
+from homeassistant.core import HomeAssistant
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.event import track_time_interval
+from homeassistant.helpers.typing import ConfigType
 
 _LOGGER = getLogger(__name__)
 
@@ -40,11 +42,11 @@ CONFIG_SCHEMA = vol.Schema(
 )
 
 
-def setup(hass, config):
+def setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Feedreader component."""
-    urls = config.get(DOMAIN)[CONF_URLS]
-    scan_interval = config.get(DOMAIN).get(CONF_SCAN_INTERVAL)
-    max_entries = config.get(DOMAIN).get(CONF_MAX_ENTRIES)
+    urls = config[DOMAIN][CONF_URLS]
+    scan_interval = config[DOMAIN].get(CONF_SCAN_INTERVAL)
+    max_entries = config[DOMAIN].get(CONF_MAX_ENTRIES)
     data_file = hass.config.path(f"{DOMAIN}.pickle")
     storage = StoredData(data_file)
     feeds = [

--- a/tests/components/feedreader/test_init.py
+++ b/tests/components/feedreader/test_init.py
@@ -3,7 +3,7 @@ from datetime import timedelta
 from os import remove
 from os.path import exists
 from unittest import mock
-from unittest.mock import patch
+from unittest.mock import mock_open, patch
 
 import pytest
 
@@ -32,6 +32,13 @@ def load_fixture_bytes(src):
     feed_data = load_fixture(src)
     raw = bytes(feed_data, "utf-8")
     return raw
+
+
+@pytest.fixture(scope="module", autouse=True)
+def mock_pickle_open():
+    """Mock builtins.open for feedreader storage."""
+    with patch("homeassistant.components.feedreader.open", mock_open(), create=True):
+        yield
 
 
 @pytest.fixture(name="feed_one_event")

--- a/tests/components/feedreader/test_init.py
+++ b/tests/components/feedreader/test_init.py
@@ -1,7 +1,5 @@
 """The tests for the feedreader component."""
 from datetime import timedelta
-from os import remove
-from os.path import exists
 from unittest import mock
 from unittest.mock import mock_open, patch
 
@@ -32,13 +30,6 @@ def load_fixture_bytes(src):
     feed_data = load_fixture(src)
     raw = bytes(feed_data, "utf-8")
     return raw
-
-
-@pytest.fixture(scope="module", autouse=True)
-def mock_pickle_open():
-    """Mock builtins.open for feedreader storage."""
-    with patch("homeassistant.components.feedreader.open", mock_open(), create=True):
-        yield
 
 
 @pytest.fixture(name="feed_one_event")
@@ -72,14 +63,10 @@ async def fixture_events(hass):
 
 
 @pytest.fixture(name="feed_storage", autouse=True)
-def fixture_feed_storage(hass):
-    """Create storage account for feedreader."""
-    data_file = hass.config.path(f"{feedreader.DOMAIN}.pickle")
-
-    yield
-
-    if exists(data_file):
-        remove(data_file)
+def fixture_feed_storage():
+    """Mock builtins.open for feedreader storage."""
+    with patch("homeassistant.components.feedreader.open", mock_open(), create=True):
+        yield
 
 
 async def test_setup_one_feed(hass):

